.github/copilot-instructions.md

```diff
@@ -103,6 +113,46 @@ uv.lock            # Locked dependency versions (auto-managed)
 - The Client class uses session management - understand the authentication flow
 - Always test with the existing test patterns to maintain consistency
 
+## Pull Request Management
+
+### PR Description and Progress Tracking
+- **Always update** the PR description throughout the development lifecycle, not just at the initial plan
+- **Update PR descriptions** when:
+  - Completing meaningful units of work (check off completed tasks)
+  - Changing approach or discovering new requirements
+  - Responding to code review feedback
+  - Adding or removing scope from the original plan
+  - Encountering blockers or dependencies
+- **Maintain a clear checklist** in PR descriptions showing:
+  - [x] Completed items with checkmarks
+  - [ ] Remaining work with clear descriptions
+  - Current status and any blockers
+  - Links to related issues or dependencies
+
+### Code Review Response Workflow
+- **When receiving code review feedback:**
+  - Update the PR description to acknowledge the feedback
+  - Add new checklist items for addressing review comments
+  - Update progress as you implement suggested changes
+  - **Always** run `uv run bin/format` and `uv run bin/test` after making changes
+  - Mark review items as completed in the PR description when addressed
+- **Keep reviewers informed** by updating the PR description with:
+  - Summary of changes made in response to feedback
+  - Any questions or clarifications needed
+  - Testing results after implementing changes
+
+### Iterative Development Communication
+- **Throughout development:**
+  - Update PR descriptions to reflect current understanding of the problem
+  - Add context about design decisions or trade-offs discovered
+  - Document any changes to the original scope or approach
+  - Keep the checklist current - add new items as they're discovered
+  - Remove or modify items that are no longer relevant
+- **Before requesting review:**
+  - Ensure PR description accurately reflects all work completed
+  - Include testing evidence and verification steps
+  - Highlight any areas where reviewer input is specifically needed
+
```

Comment:
Update instructions:

Make the minimal amount of changes to complete the task fully, without adding unnecessary complications or premature optimisations.
Simple is better than complex.
Only add comments when the code is not idiomatic. The first preference should always be writing code which is self explanatory without comments.
Tests should test intended functionality rather than just testing the specifics of the implementation. Tests should not require updates when making trivial changes to logic, only when new features are added or behaviours change in a way that we want to test for regressions in the future.